Doc/whatsnew/3.10.rst

```diff
@@ -793,6 +793,11 @@ Added :data:`~os.O_EVTONLY`, :data:`~os.O_FSYNC`, :data:`~os.O_SYMLINK`
 and :data:`~os.O_NOFOLLOW_ANY` for macOS.
 (Contributed by Dong-hee Na in :issue:`43106`.)
 
+Changed :func:`os.path.splitdrive()` on Windows to rely on a native API.
+This has improved support for special prefixes, but may have changed the
+results for some invalid UNC paths.
+(Contributed by Steve Dower in :issue:`43105`.)
```

Comment:
The ntpath changes are no longer in this PR. 

Should it be noted that, starting with 3.8, setting drive-relative paths for a different drive than the process working directory in `sys.path` no longer works for loading extension modules, such as the example I demonstrated with `sys.path.insert(0, 'C:test37')` and a process working directory on drive "F:". I don't think that anyone was ever relying on the drive-relative case, regardless of whether it worked prior to 3.8, so maybe this change can just pass silently. The rooted case was more important to handle because POSIX programmers assume "/spam" is an absolute path, and this PR keeps that case consistent with 3.7.

Reply:
Yeah, I think that one can pass silently, given how hard it is to construct working drive relative paths for anything at all.